Lib/test/test_asyncore.py

```diff
@@ -800,6 +801,36 @@ def test_quick_connect(self):
             if t.is_alive():
                 self.fail("join() timed out")
 
+    def test_map_altered_in_loop(self):
+        family = self.family
+        fail = self.fail
+
+        class NoWriteClient(BaseClient):
+
+            def handle_write(self):
+                fail("This is a non writable socket")
+
+            def writable(self):
+                return False
+
+        class ManagerClient(BaseClient):
+
+            def __init__(self, family, address):
+                BaseClient.__init__(self, family, address)
+                self.to_close = BaseClient(family, address)
+
+            def handle_write(self):
+                self.to_close.close()
+                # This trusts that the fd of this client new
+                # will be the same to the one just closed
+                new_client = NoWriteClient(family, server.address)
```

Comment:
If there is another thread running, we may not get the same fd - in this case we should skip the test with a warning.

The comment should explain that this test is relevant only if the new client got the same fd of the old client, and in the code we can verify this:

    new_client = ...
    if new_client.fileno() != old_fd:
        skip test with a warning...